### ISSUE TITLE
[Bug] `vue/no-ref-as-operand` is not thrown

### ISSUE THREAD

**Issue by wenfangdu** [vuejs/eslint-plugin-vue#1563]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 6.8.0
- **eslint-plugin-vue version:** 7.13.0
- **Node version:** 16.3.0
- **Operating System:** Windows 10

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: ['plugin:vue/vue3-essential', 'eslint:recommended', '@vue/prettier'],
  parserOptions: {
    parser: 'babel-eslint',
  },
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
  },
}
```

**What did you do?**

```js
import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import store from './store'

createApp(App).use(store).use(router).mount('#app')

// 👇 should've thrown
router.currentRoute + 1
```

**What did you expect to happen?**
Throw error.

**What actually happened?**
No error is thrown.

**Repository to reproduce this issue**
https://github.com/wenfangdu/eslint-plugin-vue-repro/tree/vue/no-ref-as-operand

**Comment by ota-meshi** [comment]
ESLint checks using only the AST of the file to be checked.
That is, `vue/no-ref-as-operand` rule can only check variables generated using `ref()` in that file.
If you want to check the type strictly, use TypeScript.

Thank you for your understanding.

**Comment by Holi0317** [comment]
Is there any rules I can use for checking ref unwrapping if I have a typescript project, configured eslint typescript with type information?
